urlupload.php

```diff
@@ -50,6 +55,15 @@ public function uploadFromUrl()
     {
     	throw new Exception("Url was not able to be opened");
     }
+
+    // we need to remember the attachment filename.  requests by rn comes with
+    // sn_status_st96.xml inside the zip.  
+    // @TODO: find a more elegant way to do this!
```

Comment:
**No change needed**, but maybe in the future we can use a hash function to name trademark xml data?  For example:

`$localname = MD5($trademarkName)` (using the name of the trademark as an exmple)

This will generate a hash string which we can use to lookup the trademark data in the file system rather than using regex (preg_match) to find/name it. This is more performant, but it depends on a consistent input coming from USPTO

Food for thought.

Reply:
The problem is that the status file inside the zip file contains the serial number in the file name even on a request by registration number.  If the user entered the registration number we wouldn't know the serial number or the status file name.  The attachment header returned by the uspto is serialnumber.zip.  I'm scraping the serial number from header which seems inelegant but I didn't see a better option.